runtime/drivers/duckdb/duckdb.go

```diff
@@ -788,7 +841,7 @@ func (c *connection) detachAllDBs() {
 		}
 
 		db := dbName(entry.Name(), version)
-		_, err = c.db.ExecContext(context.Background(), fmt.Sprintf("DETACH %s", safeSQLName(db)))
+		_, err = conn.ExecContext(ctx, fmt.Sprintf("DETACH %s", safeSQLName(db)))
```

Comment:
Should we also add some timeout to `detach` with fatal log for timeout?

Reply:
Good idea, added a timeout for the close. But skipping for the reopen for now since not quite sure how long certain checkpoints can take during normal operation.

Also not sure if a go-side timeout will matter much, since if DuckDB hangs it tends not to yield back control to Go to check the cancellation....